tensorflow_addons/image/interpolate_spline_test.py

```diff
@@ -0,0 +1,365 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for interpolate_spline."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+from scipy import interpolate as sc_interpolate
+
+import tensorflow as tf
+import tensorflow.compat.v1 as tf1  # TODO: locate placeholder
+from tensorflow.python.training import momentum
+from tensorflow_addons.utils import test_utils
+from tensorflow_addons.image import interpolate_spline
+
+
+class _InterpolationProblem(object):
+    """Abstract class for interpolation problem descriptions."""
+
+    def get_problem(self, optimizable=False, extrapolate=True,
+                    dtype='float32'):
+        """Make data for an interpolation problem where all x vectors are n-d.
+
+        Args:
+        optimizable: If True, then make train_points a tf.Variable.
+        extrapolate: If False, then clamp the query_points values to be within
+        the max and min of train_points.
+        dtype: The data type to use.
+
+        Returns:
+        query_points, query_values, train_points, train_values: training and
+        test tensors for interpolation problem
+        """
+
+        # The values generated here depend on a seed of 0.
+        np.random.seed(0)
+
+        batch_size = 1
+        num_training_points = 10
+        num_query_points = 4
+
+        init_points = np.random.uniform(
+            size=[batch_size, num_training_points, self.DATA_DIM])
+
+        init_points = init_points.astype(dtype)
+        train_points = (tf.Variable(init_points)
+                        if optimizable else tf.constant(init_points))
+        train_values = self.tf_function(train_points)
+
+        query_points_np = np.random.uniform(
+            size=[batch_size, num_query_points, self.DATA_DIM])
+        query_points_np = query_points_np.astype(dtype)
+        if not extrapolate:
+            query_points_np = np.clip(query_points_np, np.min(init_points),
+                                      np.max(init_points))
+
+        query_points = tf.constant(query_points_np)
+        query_values = self.np_function(query_points_np)
+
+        return query_points, query_values, train_points, train_values
+
+
+class _QuadraticPlusSinProblem1D(_InterpolationProblem):
+    """1D interpolation problem used for regression testing."""
+    DATA_DIM = 1
+    HARDCODED_QUERY_VALUES = {
```

Comment:
I guess my preference is to leave them where they are so that the the tests and the test constants are co-located...also that's how the code was in TF.